### PR TITLE
ocaml-lsp, lsp, jsonrpc: allow overriding the source globally

### PR DESCRIPTION
**Pull request by anmonteiro**
###### Motivation for this change

It wasn't possible to override the `src` for these packages so that they'd affect each other (they're released in tandem).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+, jsonrpc
+, omd
+, octavius
+, dune-build-info
+, uutf
+, csexp
+, cmdliner
+}:
+
+buildDunePackage {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix:14]: Missing a meta set.

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+, stdlib-shims
+, ppx_yojson_conv_lib
+, ocaml-syntax-shims
+, yojson
+, result
+, fetchzip
+}:
+
+
+buildDunePackage rec {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:11]: Missing a meta set.

File: pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix:14]: You can often do something like

```nix
  meta = jsonrpc.meta // {
    description = "…";
    # other differing fields…
  };
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Otherwise lgtm, seems like a fairly mechanical change.

**Comment by anmonteiro** [comment]
@sternenseemann thanks for the review! I applied your suggestions.

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  useDune2 = true;
+  minimumOCamlVersion = "4.06";
+
+  buildInputs = [ yojson stdlib-shims ocaml-syntax-shims ppx_yojson_conv_lib result ];
+
+  meta = with lib; {
+    description = "Jsonrpc protocol implementation in OCaml";
+    license = lib.licenses.isc;
+    platforms = platforms.unix;
+    maintainers = [ maintainers.symphorien maintainers.marsam ];
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:29]:
```suggestion
    maintainers = with maintainers; [ symphorien marsam ];
```

File: pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  useDune2 = true;
+  minimumOCamlVersion = "4.06";
+
+  buildInputs = [ yojson stdlib-shims ocaml-syntax-shims ppx_yojson_conv_lib result ];
+
+  meta = with lib; {
+    description = "Jsonrpc protocol implementation in OCaml";
+    license = lib.licenses.isc;
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/ocaml-lsp/jsonrpc.nix:27]:
```suggestion
    license = licenses.isc;
```

File: pkgs/development/ocaml-modules/ocaml-lsp/default.nix [hunk 5]
```diff
@@ -1,78 +1,15 @@
...
   useDune2 = true;
 
-  inherit preBuild;
+  inherit (lsp) preBuild;
 
-  buildInputs = buildInputs ++ [ lsp ];
+  buildInputs = lsp.buildInputs ++ [ lsp ];
 
-  meta = with lib; {
+  meta = with lib; jsonrpc.meta // {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/ocaml-lsp/default.nix:12]: Minor nit, `with lib;` is not necessary here.

File: pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    octavius
+    uutf
+    csexp
+    dune-build-info
+    omd
+    cmdliner
+    jsonrpc
+  ];
+
+  meta = with lib; jsonrpc.meta // {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/ocaml-lsp/lsp.nix:42]: Same here.

**Review by sternenseemann — APPROVED** [review]

File: pkgs/development/ocaml-modules/ocaml-lsp/default.nix (hunk 5 above)
Comment (anmonteiro) [reply pkgs/development/ocaml-modules/ocaml-lsp/default.nix]: right again you are. Fixed!

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/ocaml-lsp/default.nix]: A squash and this should be good to go I think :)

**Comment by sternenseemann** [comment]
@ofborg build ocamlPackages.ocaml-lsp

**Comment by anmonteiro** [comment]
Thanks both for the review! I just squashed my commits.

**Comment by symphorien** [comment]
tested with ocaml 4.07 and it works, thanks

**Pull request merged**